### PR TITLE
Don't require admin authentication when waiting for remote instance

### PR DESCRIPTION
**Pull request by mikedickey**
Updated wait_for_splunk_instance.yml so that the HTTP request it makes does not require authentication. When waiting for a remote instance, this created a dependency that the remote instance must have the same admin password as the local instance, which likely doesn't hold true outside of crafted test environments. Splunk itself has no such requirement, which is why it has and uses pass4SymmKey secrets.

Only the root endpoint appears to be unauthenticated in the REST API, so this prevents us from being able to query uptime of the remote instance and ensure it is > 10 seconds. I don't know why this check was put in place, but generally artificial wait times like this can be better handled via more specific checks and retry loops.

### CODE REVIEW

File: roles/splunk_common/tasks/wait_for_splunk_instance.yml [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
-    url: "{{ cert_prefix }}://{{ splunk_instance_address }}:{{ splunk.svc_port }}/services/server/info?output_mode=json"
+    url: "{{ cert_prefix }}://{{ splunk_instance_address }}:{{ splunk.svc_port }}"
     method: GET
-    user: "{{ splunk.admin_user }}"
-    password: "{{ splunk.password }}"
     validate_certs: false
   register: task_response
   until:
     - task_response.status == 200
-    - lookup('pipe', 'date +"%s"')|int - task_response.json.entry[0].content.startup_time > 10
```
Comment (nwang92) [inline roles/splunk_common/tasks/wait_for_splunk_instance.yml]: cc @arctan5x do you remember what this check was for? I wonder if it's safe to remove now that most of this provisioning is done asynchronously

**Comment by nwang92** [comment]
In general, looks good to me. I wonder if there are any plays or tasks that require knowing a remote instance's admin password? 

If this is the only situation this occurs, then maybe we can close out https://github.com/splunk/splunk-ansible/issues/206?

File: roles/splunk_common/tasks/wait_for_splunk_instance.yml (hunk 1 above)
Comment (arctan5x) [reply roles/splunk_common/tasks/wait_for_splunk_instance.yml]: Splunk used to restart couple times through Ansible provisioning and wanted to give some time after Splunk starts to stabilize.

**Comment by mikedickey** [comment]
> In general, looks good to me. I wonder if there are any plays or tasks that require knowing a remote instance's admin password?
> 
> If this is the only situation this occurs, then maybe we can close out #206?

This came up via work on breaking down CRDs into smaller components for our K8s operator. So far, I've identified only two cases where component A requires an authentication secret for component B:

* idxc.pass4SymmKey is required for search heads to communicate with cluster masters / indexers
* pass4SymmKey (the top level one) is required by any instance to register with a license master.

This of course excludes the use of pass4SymmKey secrets inside the context of components, like search head cluster members communicating with captain, index cluster members communicating with cluster master, etc.

I haven't encountered any case where an instance has to have the same admin password as any other instance. I believe the pass4SymmKey secrets are always used instead.

**Pull request merged**